docs/spec/spec.md

```diff
@@ -16,37 +16,49 @@ This document details our _Spec_ and _Status_ customizations.
+| `filter`     | [`TriggerFilter`](#triggerfilter)           | Optional    | The filter to apply against all events from the broker. Only events that pass this filter will be sent to the subscriber. | subscribe to all events |
+| `subscriber` | [`duckv1.Destination`](#duckv1.destination) | Required    | The addressable that receives events from the broker that pass the filter.                                                |                         |
 
 #### Status
 
-| Field              | Type       | Description                                                               | Constraints |
-| ------------------ | ---------- | ------------------------------------------------------------------------- | ----------- |
-| observedGeneration | int64      | The 'Generation' of the Broker that was last processed by the controller. |             |
-| conditions         | Conditions | Trigger conditions.                                                       |             |
-| subscriberURI      | apis.URL   | Address of the subscribing endpoint.                                      |             |
+The `TriggerStatus` represents the current state of the `Trigger`.
+
+| Field Name           | Field Type                            | Requirement | Description                                                                            | Constraints |
+| -------------------- | ------------------------------------- | ----------- | -------------------------------------------------------------------------------------- | ----------- |
+| `observedGeneration` | `int`                                 | Optional    | The 'Generation' of the `Trigger` that was last processed by the controller.           |             |
+| `conditions`         | [`[]apis.Condition`](#apis.condition) | Optional    | Trigger conditions. The latest available observations of the resource's current state. |             |
+| `annotations`        | `map[string]string`                   | Optional    | Fields to save additional state as well as convey more information to the user.        |             |
+| `subscriberURI`      | [`apis.URL`](#apis.url)               | Required    | The resolved address of the receiver for this trigger.                                 |             |
 
 ##### Conditions
 
-- **Ready.** True when the Trigger is provisioned and configuration is ready to
+- **Ready.** True when the trigger is provisioned and configuration is ready to
   deliver events to the subscriber.
-- **BrokerExists.** True when the Broker exists and is ready.
-- **Subscribed.** True when the subscriber is subscribed to the Broker.
+- **BrokerReady.** True when the broker exists and is ready.
+- **SubscriptionReady.** True when the subscriber is subscribed to the broker.
+- **DependencyReady.** True when the sources the trigger depends on are ready.
+- **SubscriberResolved.** True when the subscriber is resolved, i.e. the trigger
+  can get its address.
```

Comment:
Are these conditions part of the spec too? I would expect some of them, such as DependencyReady, might not be relevant in some system. If it is required, it should be set to `True` if not applicable. If some are optional, it will worth a table like the Status.

Reply:
I wonder if for the Spec we should only define the top level Ready condition that says if the Trigger is ready. For example Brokers have different kinds of conditions, would make sense for Trigger to behave the same way. But requiring the top level Ready condition should be a requirement imho.

Reply:
I agree that keeping only the **Ready** condition here would be preferable. Since the other conditions may be useful from an implementation point of view, I can move them to a separate document that also includes the events.